app/routes.py

```diff
@@ -1 +1,274 @@
-from flask import Blueprint
\ No newline at end of file
+from os import abort
+from app import db
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+from sqlalchemy import asc, desc
+from datetime import datetime
+# from slack_sdk import WebClient
+# from slack_sdk.errors import SlackApiError
+import os
+import requests
+from app.models.goal import Goal
+import json
+
+
+
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"message":f"task {model_id} invalid"}, 400))
+
+    model = cls.query.get(model_id)
+    
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+    
+    return model
+
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    try:
+        new_task = Task.from_dict(request_body)
+
+    except:
+        abort(make_response({"details": "Invalid data"}, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    
+    return make_response(jsonify({"task": new_task.to_dict()}),201)
+
+@tasks_bp.route("", methods=["GET"])
+
+def read_all_tasks():
+    title_query = request.args.get("title")
+    sort_filter = request.args.get("sort")
+    
+    if title_query:
+        tasks = Task.query.filter_by(title=title_query)
+    else:
+        tasks = Task.query.all()
+
+    if sort_filter:
+        if sort_filter == "asc":
+            tasks = Task.query.order_by(Task.title.asc())
+        elif sort_filter == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    
+    tasks_response = []
+    for task in tasks:
+        tasks_response.append(task.to_dict())
+    
+    return jsonify(tasks_response)
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+
+def read_one_task(task_id):
+    task = validate_model(Task, task_id)
+    if task.goal_id:
+        return {"task": task.to_dict_with_goal()}
+    else:
+        return {"task": task.to_dict()}
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
```

Comment:
Right now, if a user sends a request without the keys title or description your server would crash. There's a couple of ways to handle this, you could call the `validate_request` function before you access the keys in `request_body` or you could implement a `try/except` block.